crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -452,22 +456,23 @@ impl InstancePool {
 
         // Decommit any linear memories that were used
         for (memory, base) in instance.memories.values_mut().zip(self.memories.get(index)) {
-            let memory = mem::take(memory);
+            let mut memory = mem::take(memory);
             debug_assert!(memory.is_static());
 
             // Reset any faulted guard pages as the physical memory may be reused for another instance in the future
             #[cfg(all(feature = "uffd", target_os = "linux"))]
             memory
                 .reset_guard_pages()
                 .expect("failed to reset guard pages");
+            drop(&mut memory); // require mutable on all platforms, not just uffd
```

Comment:
Oh this was a weird one where `memory.reset_guard_pages()` required a `mut` binding on `memory` but that call only happens with uffd/linux, so other platforms would get a warning about an unused `mut`. To fix that I just unconditionally added a line that would require `mut memory` here regardless of the platform so this would compile everywhere without warnings.

Reply:
Ah, that makes complete sense.